### ISSUE TITLE
Proposal：Add empty states to some pages

### ISSUE THREAD

**Issue by HEREYUA** [go-gitea/gitea#29810]
### Feature Description

When users enter without any action, the homepage presents the following effect:

![image](https://github.com/go-gitea/gitea/assets/37935145/07ac2b8d-4ad5-4307-b3f9-9657822fe77a)

The activity page of personal information is as follows:

![image](https://github.com/go-gitea/gitea/assets/37935145/c2bf08b3-9b2e-4b00-849d-efad1b0da823)

Is it possible to add some empty states to these pages, like this：

When users enter the homepage, they can see some prompt information

![image](https://github.com/go-gitea/gitea/assets/37935145/4518f74d-11f4-412b-a964-36791d3cd628)

When a user has not set to hide page activity and there is no content, display the empty state.

![image](https://github.com/go-gitea/gitea/assets/37935145/abccaac6-fde5-493e-a38c-81909c562e34)

If there are no issues, I will submit the related PR

**Comment by silverwind** [comment]
Sounds good, we can introduce a "blankslate" shared template like this:

<img width="977" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/94053474-7ffa-43ae-9a96-3039110fbc93">

Ideas:

- In some cases it makes sense to render a background, in some not, so I would add a option `Background=true/false` to the template
- Don't add a icon, at least not by default. Have an `Icon` option if needed in some places.

**Comment by denyskon** [comment]
We already have those kinds of templates for e.g. packages or wiki which look nice

**Comment by lunny** [comment]
> Idea for arguments to the shared template:
> 
>     * Icon=string - svg name, default empty string means no icon
> 
>     * Title=string - title string or HTML template
> 
>     * Description=string - description string or HTML template
> 
>     * Classes=string - additional classes
> 
>     * Background=boolean - whether to render background and border

I don't think we should share them. Most of them should be a short code and have different content and conditions. It's unnecessary to share them.

**Comment by silverwind** [comment]
> I don't think we should share them. Most of them should be a short code and have different content and conditions. It's unnecessary to share them.

We have to at least share a CSS class for common styling.

**Comment by lunny** [comment]
> > I don't think we should share them. Most of them should be a short code and have different content and conditions. It's unnecessary to share them.
> 
> We have to at least share a class for common styling, or do it via tailwind which would make a shared template more ideal to keep the elements in-sync.
> 
> I'm still in favor of a shared template, we have to reduce all the copy-paste in templates.

If the template is too short. The invoke code maybe the same size as the shared template. It introduced new knowledge but maybe not bring obvious benefits.
